gcc/rust/backend/rust-compile-intrinsic.cc

```diff
@@ -34,6 +34,22 @@
 namespace Rust {
 namespace Compile {
 
+static bool
+is_basic_integer_type (TyTy::BaseType *type)
+{
+  switch (type->get_kind ())
+    {
+    case TyTy::INT:
+    case TyTy::UINT:
```

Comment:
Do these cover all u* and i* sizes other than usize and isize?

Reply:
Yes, they do. This is similar to [`rustc`'s`TyKind` enum](https://doc.rust-lang.org/stable/nightly-rustc/rustc_middle/ty/enum.TyKind.html) except we can't have sum types, so we're using inheritance to counter that. All of our `TyTy` nodes inherit from the `BaseType` class, which exposes `get_kind()`. Then specifically the `IntType` and `UintType` classes (which inherit from `BaseType`) expose one new enum each and a `get_{u}int_kind()` method